### ISSUE TITLE
Error Cannot find module '/Users/XXX/.taco_home/remote-builds/taco-remote/builds/766/cordovaApp/

### ISSUE THREAD

**Issue by jeromeXoo** [Microsoft/TACO#294]
Hi,

I use Visual Studio 2015, and when i launch a remotebuild on mac, i have the following error:
```
1>  Executing "before_prepare"  hook for all plugins.
1>  Running command: /Users/Jerome/.taco_home/remote-builds/taco-remote/builds/766/cordovaApp/hooks/before_prepare/01_jshint.js /Users/Jerome/.taco_home/remote-builds/taco-remote/builds/766/cordovaApp
1>  module.js:327
1>      throw err;
1>      ^
1>
1>  Error Cannot find module '/Users/Jerome/.taco_home/remote-builds/taco-remote/builds/766/cordovaApp/
1>  '
1>      at Function.Module._resolveFilename (module.js:325:15)
1>      at Function.Module._load (module.js:276:25)
1>      at Function.Module.runMain (module.js:441:10)
1>      at startup (node.js:139:18)
1>      at node.js:968:3
1>  Command finished with error code 1: /Users/Jerome/.taco_home/remote-builds/taco-remote/builds/766/cordovaApp/hooks/before_prepare/01_jshint.js /Users/Jerome/.taco_home/remote-builds/taco-remote/builds/766/cordovaApp
1>  Failed to build app for buildNumber 766: Hook failed with error code 1: /Users/Jerome/.taco_home/remote-builds/taco-remote/builds/766/cordovaApp/hooks/before_prepare/01_jshint.js
1>MSBUILD : cordova-build error : Error: Remote build error from the build server Build failed with error Hook failed with error code 1: /Users/Jerome/.taco_home/remote-builds/taco-remote/builds/766/cordovaApp/hooks/before_prepare/01_jshint.js - 1
```
Following the start of 01_jshints.js file:
```
#!/usr/local/bin/node 

module.exports = function (context) {
    var fs = require('fs');
    var path = require('path');
    var jshint = require('jshint').JSHINT;
    var async = require('async');
```
If anyone understands what's wrong, it'd be great.

The compilation on PC with Android as target is OK.

**Comment by MSLaguana** [comment]
What's happening here is your jshint hook is expecting jshint (and async) to be installed into the project on the mac build machine, but it is not. My understanding was that hooks should not `require()` things, but instead should use something like `var Q = context.requireCordovaModule('q');`. However that only applies to modules that are used by cordova itself, not additional dependencies that you add.

Have you `npm install`ed on your windows machine? I believe in VS in the solution explorer, there is a `dependencies` entry, does that reference jshint?

**Comment by jeromeXoo** [comment]
Thank you very much for your answer.

To understand: "var = Q context.requireCordovaModule ('q');" I was on the cordova site. I've realized that hooks through the directories were depreciated (eg : hooks/after_build/after_build_custom_action.js).

I deleted directories "after_prepare" and "before_prepared". I referenced the hooks in the config.xml file only. And the compilation was executed without error.

And I will replace the "require ()" with "context.require Cordova Module", which is more correct.

thank you again for your help.
